operator/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -166,6 +166,12 @@ type ClusterSpec struct {
 	// By default if Replicas is 3 or more and redpanda.default_topic_partitions is not set
 	// default webhook is setting redpanda.default_topic_partitions to 3.
 	AdditionalConfiguration map[string]string `json:"additionalConfiguration,omitempty"`
+	// Cluster configuration values may also be held here. A `keyName` entry in this
+	// attribute will override corresponding `redpanda.keyname` entries in AdditionalConfiguration;
+	// this is to permit the migration of those settings to this attribute.
+	// The configuration may contain references to values extracted from k8s ConfigMaps or Secrets;
+	// furthermore, we support the fetching of provider-specific secrets directly.
+	ClusterConfiguration ClusterConfiguration `json:"clusterConfiguration,omitempty"`
```

Comment:
This won't matter if ClusterConfiguration becomes a map type but `omitempty` doesn't work on structs.

Reply:
That's fixed now.